test/unit/tracks/tracks.test.js

```diff
@@ -342,3 +343,48 @@ if (Html5.supportsNativeTextTracks()) {
     emulatedTt.on('addtrack', addtrack);
   });
 }
+
+test('removes cuechange event when text track is hidden for emulated tracks', function() {
```

Comment:
Yeah. If you run this test without the change, `numTexttTrackChanges` reveals higher numbers (due to the old event listeners hanging around and triggering `texttrackchange` events.